### DISCUSSION TITLE
How to send a message ?

### DISCUSSION THREAD

**Discussion by vlourme** [sbertix/Swiftagram#122] [Q&A] [answered]
Hi! I'm trying to make a IGDM-like on mobile, actually I'm successfully getting my message list, messages' author, etc. But I can't answer them. 

I did not found a method to send a message, is it included yet?
Or maybe it is not included to prevent spam?

Thank you 👍

**Comment by sbertix** [comment, marked as answer]
Hey @vlourme,

Sending messages through **Swiftagram** is not supported and you're correct, it's mainly because of a concern on its uses. 
Other endpoints, or even response properties, haven't been exposed to the wrapper for that exact reason. 
That said, as all encryption features used by Instagram are included with the library, a quick googling it's basically all that's needed to have it working 👍

**Reply by vlourme** [reply]
Thank you for answering, I'll do some researches on this :)

So, you mean I can extend the library endpoints and add the `send_message` endpoint by myself?

Edit: I successfully found how to do! ComposableRequest is well made, very cool to use.

**Reply by sbertix** [reply]
Hey @vlourme, 

I've decided to ease the self-imposed restrictions I had put on sending dms. 
As of `5.0.0` anyone will be able to send text only messages to **already existing** conversations. 
I feel like this is a good enough compromise between providing the most complete API wrapper, and caution about possible misuses. 

I'm writing this here just for reference.
